lib/stitch_fix/log_weasel/middleware.rb

```diff
@@ -14,16 +15,23 @@ def initialize(app, options = {})
     # Future: Maybe add X-Amzn-Trace-Id request header for tracing through load balancer
     # (http://docs.aws.amazon.com/elasticloadbalancing/latest/application/load-balancer-request-tracing.html)
     def call(env)
+      req = Rack::Request.new(env)
       x_correlation_id = env.fetch(REQUEST_ID_KEY, nil) || env.fetch(CORRELATION_ID_KEY, nil)
 
+      if ENV.fetch('LOGWEASEL_FROM_PARAMS', nil) && req.params[PARAMS_KEY]
```

Comment:
We want to check both for the existence of the value, as well as the env var, because apps like `csp-report-service` can still receive requests at other endpoints from stitch fix apps that should follow the normal log weasel transaction setting flow.